File: src/main/java/com/spring/products/entity/Product.java
package com.spring.products.entity;

import com.spring.products.enumeration.Status;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Product {
    //@Id
    //@GeneratedValue(strategy = GenerationType.IDENTITY)
    //private long productId;
    @Id
    private String productSerialNumber;
    @Column(unique = true)
    private String productName;
    private String type;
    private String Model;
    private String specification;
    private Status status;
    @ManyToOne
    @JoinColumn(name = "fk_brand_id")
    private Brand brand;
    @ManyToOne
    @JoinColumn(name = "fk_category_id")
    private Category category;
}

File: src/app/product.ts
import { Brand } from "./brand";
import { Category } from "./category";
import { Status } from "./status.enum";

export class Product {
    productSerialNumber! : string;
    productName! : string;
    type! : string;
    model!:string;
    specification!:string;
    status!:Status;
    brand! : Brand;
    category! : Category;
}
